AaC Developer Docs Updates - ZSH Command Inclusions & Profiling AaC Package Updates (#492)
jondavid-black/AaC: docs/docs/dev_guide/profiles/profiling.md

Wrote system design document to ../../../temp/alarm_clock_system_design_document.md
```

> *The above example is the top ten result from sorting based on the time taken from the code being run in `gen_design_doc_impl.py`
> With the above example output you can see that the code that this was run against has some hints of recursion occurring. 
> Without recursion occuring the two numbers at the top would be the same.*

Also to note from the output above you can see where some of the code that is being run multiple times and is taking up more time than others. It is recommended to see the full output at least first to make sure that eveyrthing is getting hit and run. Then, as you go, whittle down to find the code that is offending more often.

## Profile Description

`profile` is another deterministic profiler. There are a lot of similaarities to the `cProfile` however, this profiler has a significant amount of overhead and this profiler is pure python based. Therefore if you are needing to extend the profiler this can be done with this package.

> It is recommended to use `cProfile` since it can cover the majority of use-cases, `profile` is when there is more overhead needed and the extra extensibility can be utiliized. You can find the `cProfile` documentation here: [cProfile Setup and Justification](./cprofile.md)

Reviewer: Re-organized the flow a smidge.
```suggestion
It is recommended to use `cProfile` for the majority of use-cases, and to use `profile` for extensibility and when there is less consideration for overhead or performance. You can find the `cProfile` documentation here: [cProfile Setup and Justification](./cprofile.md)
```

>*_NOTE: This profiler is a more advanced version of `cProfile` and has more extensibility and overhead that can be utilized. Majority of cases can be achieved with the `cProfile` package. 

### Getting Started with Profile

To get started using the `profile` package you will need to import this into the plugin that is being developed, new or existing.

To do so just use the import statement to bring it in:

```python
import profile
```

To enable the `profile` package is a little more complicated than the `cProfile` conterpart.
`profile` does not support using a Context Manager so the wrapping around the activity will be a little more work:

```python

""" Run Command for profile to run an AaC command within the plugin code """
profile.run(command, filename=None, sort=- 1)

""" This command is similar to run() except that it accepts globals and locals definitions that are supplied and passed through the command being executed. """
profile.runctx(command, globals, locals, filename=None, sort=- 1)

""" The Profile() class. This is really only needed for more precise controls over the profiling being done
than what the cProfile.run() method can provide. """

> class profile.Profile(timer=None, timeunit=0.0, subcalls=True, builtins=True)

Reviewer: Oddly, I can't seem to add a suggestion with the correct code block enclosure, but you're missing a '```' at the end of this code block.
